spec/features/customers/show_spec.rb

```diff
@@ -0,0 +1,60 @@
+require "rails_helper"
+
+RSpec.describe "the custmer show page" do
+  it "displays the name of customer" do
+    market1 = Supermarket.create(name: "Supermercado", location: "123 Food St.")
+    item1 = market1.items.create(name: "Banana", price: 1)
+    item2 = market1.items.create(name: "Coffee", price: 3)
+    item3 = market1.items.create(name: "Pasta", price: 3)
+    customer1 = Customer.create(name: "Guy")
+    visit "/customers/#{customer1.id}"
+
+    expect(page).to have_content("#{customer1.name}'s Shopping List")
+  end
+
+  it "displays the list of items they intend to buy and what supermarket they are from" do
+    market1 = Supermarket.create(name: "Supermercado", location: "123 Food St.")
+    item1 = market1.items.create(name: "Banana", price: 1)
+    item2 = market1.items.create(name: "Coffee", price: 3)
+    item3 = market1.items.create(name: "Pasta", price: 3)
+    customer1 = Customer.create(name: "Guy")
+    customer_item1 = CustomerItem.create(item: item1, customer: customer1)
+    customer_item2 = CustomerItem.create(item: item2, customer: customer1)
+    customer_item3 = CustomerItem.create(item: item3, customer: customer1)
+    visit "/customers/#{customer1.id}"
+
+    within("div.grocery-list") do
```

Comment:
Nice work setting up these within blocks!

Reply:
Another option: putting a html/css class on a div *inside* the iteration you do for the `@customer.items.each`
```
  <% @customer.items.each do |item| %>
    <div class = <% "item-#{item.id}" %>>
        <p>Item: <%= item.name %></p>
        <p>Cost: <%= item.price %></p>
        <p>Store: <%= item.supermarket.name %></p></br>
   </div>
 <% end %>

Now you have a div with a specific item ID so you can do `within("div.item-#{item1.id}")` and get even more granular with your assertions!